src/components/HistoryTable/HistoryTable.tsx

```diff
@@ -52,13 +53,28 @@ const HistoryTable: React.VFC<HistoryTableProps> = ({
     }
   }, [processedCommitCount]);
 
+  // This useEffect handles the jump to row functionality
+  // When we have a selected commit that we have not yet scrolled to we need to scroll to that row
   useEffect(() => {
-    if (selectedCommit && selectedCommit.rowIndex && listRef.current) {
+    if (
+      selectedCommit &&
+      selectedCommit.rowIndex &&
+      !selectedCommit.visited &&
+      listRef.current
+    ) {
       listRef.current.scrollToItem(selectedCommit.rowIndex, "center");
-    } else {
+      // We mark the selected commit as visited so that we don't keep trying to scroll to it.
+      markSelectedRowVisited();
+      // If we have not loaded the commit yet we need to keep fetching rows until we do.
+    } else if (selectedCommit && !selectedCommit.loaded) {
       loadMoreItems();
     }
```

Comment:
It might be more concise if we used a guard here

```
if (selectedCommit) {
    if (selectedCommit.rowIndex &&...) { // stuff...}
    else if (!selectedCommit.loaded) { //stuff...}
}
```

In this case I think you can also extract the fields you want like `const { rowIndex, visited, loaded } = selectedCommit;` and use those in the conditionals, instead of repeating `selectedCommit` in multiple places